app/controllers/cohorts_controller.rb

```diff
@@ -110,6 +111,11 @@ def show
           end
           options
         end
+
+        # included so the excel download can be regenerated when the visible columns change
+        params['visible_columns'] ||= @visible_columns.map { |m| m.column.to_s }.join(',')
+        # included so the excel download can be regenerated if the total number of clients changes
+        params['client_count'] ||= @cohort.search_clients(population: params[:population], user: current_user).count
```

Comment:
Maybe a simple MD5 to keep these small, what do you think?  The idea is simply to force a rebuild if anything has changed.
I'll test:
```suggestion
        # included so the excel download can be regenerated if the total number of clients changes
        params['cache_key'] ||= Digest::MD5.hexdigest(@visible_columns.map { |m| m.column.to_s }.join(',') << @cohort.search_clients(population: params[:population], user: current_user).count)
```

Reply:
This makes sense to me and looks much nicer in the database. It's also working in my tests.